types/precison_timestamp_types.go

```diff
@@ -0,0 +1,137 @@
+package types
+
+import (
+	"fmt"
+	"github.com/cockroachdb/errors"
+	"github.com/substrait-io/substrait-go/proto"
+)
+
+type TimePrecision int
+
+const (
+	UnknownPrecision TimePrecision = iota
+	Seconds
+	Milliseconds
+	Microseconds
+	Nanoseconds
+)
+
+func timePrecisionToProtoVal(val TimePrecision) int32 {
+	switch val {
+	case Seconds:
+		return 0
+	case Milliseconds:
+		return 3
+	case Microseconds:
+		return 6
+	case Nanoseconds:
+		return 9
+	}
+	panic("unreachable")
+}
+
+func ProtoToTimePrecision(val int32) (TimePrecision, error) {
+	switch val {
+	case 0:
+		return Seconds, nil
+	case 3:
+		return Milliseconds, nil
+	case 6:
+		return Microseconds, nil
+	case 9:
+		return Nanoseconds, nil
+	}
+	return UnknownPrecision, errors.Newf("invalid TimePrecision value %d", val)
```

Comment:
I believe 0-9 are all allowed. Why throw here? Are you making the design decision that substrait-go won't support 1,2,4,5,7,8? If so, maybe update methods to say that in the error.

Reply:
fixed it